lib/model/CAnomalyDetectorModelConfig.cc

```diff
@@ -623,7 +623,7 @@ CAnomalyDetectorModelConfig::factory(int identifier,
     }
     result->scheduledEvents(m_ScheduledEvents);
 
-    return result;
+    return std::move(result);
```

Comment:
It's best not to move return values.

If you _don't_ specify `move` then the compiler will choose what it thinks is best from:

1. Copy elision
2. RVO
3. Move

If you specify `move` then you're forcing option 3, which might not be the most efficient in all circumstances.

See also https://stackoverflow.com/a/17473874

Reply:
For context, this change was in response to the compiler warning ```CAnomalyDetectorModelConfig.cc:626:12: warning: prior to the resolution of a defect report against ISO C++11, local variable 'result' would have been copied despite being returned by name, due to its not matching the function return type ('shared_ptr<const ml::model::CModelFactory>' vs 'shared_ptr<ml::model::CModelFactory>') [-Wreturn-std-move-in-c++11]
    return result;
           ^~~~~~
CAnomalyDetectorModelConfig.cc:626:12: note: call 'std::move' explicitly to avoid copying on older compilers
    return result;
           ^~~~~~
           std::move(result)
1 warning generated.```

which is just a minor annoyance.. I'll revert the change.

Reply:
In https://reviews.llvm.org/rG321f24ec42f65403134031bbe99ad3c468b01efe it says:

> Diagnose cases of "return x;" or "throw x;" which in this version of Clang *do* produce moves, but which prior to Clang 3.9 / GCC 5.1 produced copies instead. This is useful in codebases which care about portability to those older compilers.

We are currently using Clang 3.9 / GCC 7.3 in `master` and `7.x`, so should be OK.  Realistically we'll never downgrade for the more recent branches.  The branch that could benefit is `6.8` where we're using Clang 3.8 / GCC 6.2, but not sure it's worth the effort of a PR for just one place.